pkg/kds/v2/client/kds_client.go

```diff
@@ -74,14 +75,18 @@ func NewKDSSyncClient(
 	}
 }
 
-func (s *kdsSyncClient) Receive() error {
+func (s *kdsSyncClient) SendReq() error {
```

Comment:
TBH I'm not sure if this fix is correct. All you did is moved `SendReq` to a separate goroutine, but this function runs once at the start and sends at most 20 DeltaDiscoveryRequests and exits (see it just loops over `s.resourceTypes`). 

However, the second function `ReceiveResp` despite the name actually not only receives DeltaDiscoveryResponses but also sends DeltaDiscoveryRequests (in NACK and ACK functions).

So it doesn't look like this PR actually decouples KDS grpc client `Send` and `Recv` methods in different goroutines.

Reply:
We need to make sure the ACK/NACK is sent after the subscribing requests.

Reply:
> We need to make sure the ACK/NACK is sent after the subscribing requests.

It's fine since the ACK/NACK needs to carry the `resourceType` parameter, see: [ACK](https://github.com/kumahq/kuma/blob/6ffe623803707cb56e866630e6eef276b7d295e5/pkg/kds/v2/client/stream.go#L119), [NACK](https://github.com/kumahq/kuma/blob/2.12.0/pkg/kds/v2/client/stream.go#L137)

In my new pseudocode, actually the `ACK/NACK` won't impact the `deltaDiscoveryReq` since they're belong to different resources types.